File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    queue =  [(i,p) for i,p in enumerate(priorities)]
    answer =0 
    while queue :
        firstone = queue.pop(0)
        if any(firstone[1] < p[1] for p in queue) :
            queue.append(firstone)
        else :
            answer+=1 #작업진행
            if firstone[0] == location:
                return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;

        Queue<int []> q = new LinkedList<>();
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for (int i = 0 ; i<priorities.length; i++)
        {
            q.add(new int [] {priorities[i],i}); // 값, idx
            pq.add(priorities[i]);
        }
        
        int cnt = 1;
        int targetval = priorities[location];
        while(!q.isEmpty())
        {
            int [] now = q.poll();

            
            int maxval = pq.peek();
            if (now[0] == maxval)
            {            
                if (now[0] == targetval && now[1] == location)
                {
                    //값도, idx도 맞다면 
                    return cnt;
                }
                pq.poll();
                cnt +=1;
            }
            else
            {
                q.add(now);
            }
            
        }
        
        return answer;
    }
}